### DISCUSSION TITLE
there is any idea about module development in adonisjs of recurring codes.

### DISCUSSION THREAD

**Discussion by AmirSoftEng** [adonisjs/core#4129] [Help]
How can we create module for adonisjs framework like chat module, forum module, blog module with template views, controller, models.

I mean, i just create a chat app, now other just install and use it in their system, without creating it from scratch.

any documentation about package/module development?

--------
I really grateful to all authors of adonisjs frameworks who give us such easy to use framework.

**Comment by McSneaky** [comment]
Aha, seems like I accidentally discarded my response...

Anyway, full doc about module development is incoming. Meanwhile you can reference some community tutorials
- https://adonismastery.com/courses/adonisjs-package-development
- https://adocasts.com/lessons/how-to-make-a-simple-adonisjs-package
- https://mcsneaky.ap3k.pro/posts/adonis-v5-provider/ (super-hyper basic one)

Most likely there are quite some more tutorials out there when Googleing for them. Or you can try to read some existing package source code
